File: solutions/go/difference-of-squares/2/difference_of_squares.go
// Package diffsquares contains methods for SquareOfSum and SumOfSquares calculation
package diffsquares

// SquareOfSum return square of the sum of the first natural numbers below number
func SquareOfSum(number int) int {
	sum := number * (number + 1) / 2
	return sum * sum
}

// SumOfSquares return sum of the squares of the first natural numbers below number
func SumOfSquares(number int) int {
	return number * (number + 1) * (2*number + 1) / 6
}

// Difference return difference of the SquareOfSum and SumOfSquares results for the number
func Difference(number int) int {
	return SquareOfSum(number) - SumOfSquares(number)
}

File: solutions/swift/difference-of-squares/2/Sources/DifferenceOfSquares/DifferenceOfSquares.swift
final class Squares {
    let num: Int

    init(_ num: Int) {
        self.num = num
    }

    private(set) lazy var squareOfSum: Int = {
        let square = (1...num).reduce(into: 0) { $0 += $1 }
        return square * square
    }()

    private(set) lazy var sumOfSquares: Int = {
        return (1...num).map { $0 * $0 }.reduce(into: 0) { $0 += $1 }
    }()

    private(set) lazy var differenceOfSquares: Int = {
        return squareOfSum - sumOfSquares
    }()
}
